repositories.bzl

```diff
@@ -40,6 +41,39 @@ def contrib_rules_jvm_deps():
         urls = ["https://github.com/protocolbuffers/protobuf/archive/v{}.tar.gz".format(PROTOBUF_VERSION)],
     )
 
+    # The `rules_java` major version is tied to the major version of Bazel that it supports,
+    # so this is different from the version in the MODULE file.
+    major_version = native.bazel_version.partition(".")[0]
+    if major_version == "5":
+        maybe(
+            http_archive,
+            name = "rules_java",
+            urls = [
+                "https://github.com/bazelbuild/rules_java/releases/download/5.5.1/rules_java-5.5.1.tar.gz",
+            ],
+            sha256 = "73b88f34dc251bce7bc6c472eb386a6c2b312ed5b473c81fe46855c248f792e0",
+        )
+
+    elif major_version == "6":
+        maybe(
+            http_archive,
+            name = "rules_java",
+            urls = [
+                "https://github.com/bazelbuild/rules_java/releases/download/6.5.2/rules_java-6.5.2.tar.gz",
+            ],
+            sha256 = "16bc94b1a3c64f2c36ceecddc9e09a643e80937076b97e934b96a8f715ed1eaa",
+        )
```

Comment:
Hum, doesn't work for me (bazel `6.4.0`):
```
ERROR: /path/to/external/remote_java_tools_linux/BUILD:27:23: @remote_java_tools_linux//:toolchain: missing value for mandatory attribute 'java_runtime' in 'java_toolchain' rule
ERROR: /path/to/external/remote_java_tools_linux/BUILD:32:23: @remote_java_tools_linux//:toolchain_hostjdk8: missing value for mandatory attribute 'java_runtime' in 'java_toolchain' rule
ERROR: /path/to/external/remote_java_tools_linux/BUILD:41:23: @remote_java_tools_linux//:legacy_toolchain: missing value for mandatory attribute 'java_runtime' in 'java_toolchain' rule
ERROR: /path/to/external/remote_java_tools_linux/BUILD:49:23: @remote_java_tools_linux//:legacy_toolchain_jvm8: missing value for mandatory attribute 'java_runtime' in 'java_toolchain' rule
ERROR: /path/to/external/remote_java_tools_linux/BUILD:56:23: @remote_java_tools_linux//:toolchain_vanilla: missing value for mandatory attribute 'java_runtime' in 'java_toolchain' rule
ERROR: /path/to/external/remote_java_tools_linux/BUILD:65:31: @remote_java_tools_linux//:toolchain_java8: missing value for mandatory attribute 'java_runtime' in 'java_toolchain' rule
ERROR: /path/to/external/remote_java_tools_linux/BUILD:72:31: @remote_java_tools_linux//:toolchain_java8_jvm8: missing value for mandatory attribute 'java_runtime' in 'java_toolchain' rule
ERROR: /path/to/external/remote_java_tools_linux/BUILD:65:31: @remote_java_tools_linux//:toolchain_java9: missing value for mandatory attribute 'java_runtime' in 'java_toolchain' rule
ERROR: /path/to/external/remote_java_tools_linux/BUILD:72:31: @remote_java_tools_linux//:toolchain_java9_jvm8: missing value for mandatory attribute 'java_runtime' in 'java_toolchain' rule
ERROR: /path/to/external/remote_java_tools_linux/BUILD:65:31: @remote_java_tools_linux//:toolchain_java10: missing value for mandatory attribute 'java_runtime' in 'java_toolchain' rule
ERROR: /path/to/external/remote_java_tools_linux/BUILD:72:31: @remote_java_tools_linux//:toolchain_java10_jvm8: missing value for mandatory attribute 'java_runtime' in 'java_toolchain' rule
ERROR: /path/to/external/remote_java_tools_linux/BUILD:65:31: @remote_java_tools_linux//:toolchain_java11: missing value for mandatory attribute 'java_runtime' in 'java_toolchain' rule
ERROR: /path/to/external/remote_java_tools_linux/BUILD:72:31: @remote_java_tools_linux//:toolchain_java11_jvm8: missing value for mandatory attribute 'java_runtime' in 'java_toolchain' rule
ERROR: /path/to/external/remote_java_tools_linux/BUILD:83:23: @remote_java_tools_linux//:toolchain_jdk_14: missing value for mandatory attribute 'java_runtime' in 'java_toolchain' rule
ERROR: /path/to/external/remote_java_tools_linux/BUILD:90:23: @remote_java_tools_linux//:toolchain_jdk_15: missing value for mandatory attribute 'java_runtime' in 'java_toolchain' rule
ERROR: /path/to/external/remote_java_tools_linux/BUILD:96:23: @remote_java_tools_linux//:prebuilt_toolchain: missing value for mandatory attribute 'java_runtime' in 'java_toolchain' rule
ERROR: /path/to/external/bazel_tools/tools/jdk/BUILD:171:14: Target '@remote_java_tools_linux//:prebuilt_singlejar' contains an error and its package is in error and referenced by '@bazel_tools//tools/jdk:prebuilt_singlejar_linux'
```
I seem to need `rules_java` **7.x**.